### PR TITLE
Convert docs from md to rst

### PR DESCRIPTION
**Pull request by s-makin**
## Why is this needed?
This PR solves all of our problems because...
Moving to rST and making use of the ability to include other
files will make parts of the docs more reusable, tutorials and
how-tos easier to write in a more consistent way, and easier to
maintain the set of common links that are used in more than one
page without having to update several pages each time one link
changes.

## Please Squash this PR with this commit message

```
- Add page anchors for easier referencing
- Add links.txt file for common links
- Add snippet files for common reusable text
- Standardised "how to enable" guides more
- Moved "enable pro in dockerfile" outside "enable" menu
```

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in: 

## Does this PR require extra reviews?

 - [ ] Yes
 - [X] No

### CODE REVIEW

File: docs/howtoguides/create_pro_golden_image.rst
Comment (s-makin) [inline docs/howtoguides/create_pro_golden_image.rst:15]: As discussed in standup, I found this paragraph confusingly written, particularly with reference to the "non-standard" Pro services

**Review by lucasmoura — APPROVED** [review]

**Comment by lucasmoura** [comment]
Let's just wait for #2856 to land before merging this one

**Pull request merged**
